test/models/document_test.rb

```diff
@@ -1,7 +1,25 @@
 require "test_helper"
 
 class DocumentTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test "does not accept files larger than 50 MiB" do
+    document = Document.new(title: "Test document")
+    file = mock_file(size: 50.megabytes, content_type: "application/pdf", filename: "testfile.pdf")
+    document.file.attach(io: file)
+
+    assert_not document.valid?
+  end
 end
+
+module DocumentHelper
+  # Helper method to create a mock file
+  def mock_file(size:, content_type:, filename:)
+    file_mock = StringIO.new
+    def file_mock.size
+      @size
+    end
+    file_mock.instance_variable_set(:@size, size)
+    file_mock.define_singleton_method(:original_filename) { filename }
+    file_mock.define_singleton_method(:content_type) { content_type }
+    file_mock
+  end
```

Comment:
Instead of creating our own mock system, we should use what is integrated in Rails, which is known as fixtures.

This section is pretty good at explaining how: https://guides.rubyonrails.org/v7.0/active_storage_overview.html#adding-attachments-to-fixtures

Reply:
I don't Ruby-on-Rails' fixture system has support for mock-files.
Do you want to store a 51 MiB file in our project?

Reply:
Hmm. I seem to have misunderstood the fixture system's flexibility...

While it could still work if we added a mocking lib on top. It might be better to skip this model constraint stuff as you said. 🫡